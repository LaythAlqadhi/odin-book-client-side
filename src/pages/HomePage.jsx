import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { useAuth } from '../contexts/AuthContext';

function HomePage() {
  const navigate = useNavigate();
  
  const handleContinueWithGitHub = (e) => {
    e.preventDefault();

    window.location.href = 'https://b32a7bae-6556-4da3-a848-f0e0b80bf4f0-00-36mr5e3zsor9c.janeway.replit.dev/v1/auth/github';
  }
  return (
    <main>
      <button type="button" onClick={() => navigate('auth/signup')}>Sign Up</button>
      <button type="button" onClick={() => navigate('auth/signin')}>Sign In</button>
      <button type="button" onClick={handleContinueWithGitHub}>Continue with GitHub</button>
    </main>
  );
}
export default HomePage;
