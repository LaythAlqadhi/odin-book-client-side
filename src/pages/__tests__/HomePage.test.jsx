import React from 'react';
import { render, screen, act, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import { vi } from 'vitest';
import HomePage from '../HomePage';

describe('HomePage component', () => {
  it('renders HomePage component in the document correctly', () => {
    render(
      <MemoryRouter>
        <HomePage/>
      </MemoryRouter>
    );
    
    expect(screen.container).toMatchSnapshot();
  });
});
