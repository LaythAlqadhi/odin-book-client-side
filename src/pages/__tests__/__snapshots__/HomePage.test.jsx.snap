// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage component > renders HomePage component in the document correctly 1`] = `undefined`;
